components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandlerTest.java

```diff
@@ -49,22 +54,34 @@
 /**
  * Unit test covering TokenResponseTypeHandler class
  */
-@PrepareForTest({OAuthComponentServiceHolder.class, OAuthEventInterceptor.class, IdentityUtil.class, OAuth2Util.class})
-public class TokenResponseTypeHandlerTest extends PowerMockTestCase {
-    @BeforeMethod
+@PrepareForTest({OAuthComponentServiceHolder.class, IdentityUtil.class, OAuthEventInterceptor.class, OAuth2Util.class,
+        IdentityTenantUtil.class})
+public class TokenResponseTypeHandlerTest extends PowerMockIdentityBaseTest {
+    @BeforeTest
     public void setUp() throws Exception {
         System.setProperty("carbon.home", System.getProperty("user.dir")
                 + File.separator + "target");
         PowerMockito.mockStatic(IdentityUtil.class);
         PowerMockito.when(IdentityUtil.getIdentityConfigDirPath())
                 .thenReturn(System.getProperty("user.dir") + File.separator + "src" + File.separator + "test"
                         + File.separator + "resources" + File.separator + "conf");
+
         OAuthComponentServiceHolder oAuthComponentServiceHolder
                 = PowerMockito.mock(OAuthComponentServiceHolder.class);
         OAuthEventInterceptor interceptor = PowerMockito.mock(OAuthEventInterceptor.class);
         OAuthServerConfiguration oAuthServerConfiguration = PowerMockito.mock(OAuthServerConfiguration.class);
         PowerMockito.when(oAuthComponentServiceHolder.getOAuthEventInterceptorProxy()).thenReturn(interceptor);
 
+        RealmService realmService = PowerMockito.mock(RealmService.class);
+        TenantManager manager = PowerMockito.mock(TenantManager.class);
+        PowerMockito.when(realmService.getTenantManager()).thenReturn(manager);
```

Comment:
Why not use WithRealmService annotation rather than the power mock

Reply:
Refer to https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/authz/AuthorizationHandlerManagerTest.java#L31

Reply:
This is older test i wrote just add few blocks to minimize the mock